tutorials/functional-full-stack/purescript-bridge/tutorial.md

```diff
@@ -0,0 +1,339 @@
+---
+title: Connecting a Haskell Backend to a Purescript Frontend
+published: not-yet
+ghc: 7.10.3
+purs: 0.10.5
+lts: 7.17
+libraries: purescript-bridge
+language: haskell purescript
+author-name: Javier Casas Velasco
+description: In this tutorial we will implement a way to extend the types in the Haskell backend to the Purescript frontend while maintaining consistency and simplifying communication.
+---
+# Connecting a Haskell Backend to a Purescript Frontend
+## Introduction
+At Stackbuilders we are working on a full-stack app with CollegeVine https://www.collegevine.com/ using functional languages.
+We have a Haskell backend, based on Servant, that manipulates the database and offers some endpoints to a Purescript frontend,
+that does all the React-like magic to show a really nice interface on the user's browser.
+It's great because we have advanced types, purity and all the awesome benefits that offers the Functional world.
+But not everything is perfect.
+
+### Motivation
+The problem is that we have two different codebases: one in Haskell and the other in Purescript.
+The syntax is almost the same, but that small difference means we can't directly share files.
+So we have types that describe the entities in the backend,
+and we have the same types that describe the same entities in the frontend, in different files.
+And every now and then, the backend team changes something on the backend types,
+but forgets to change it on the front end types.
+Suddenly we start to get the dreaded runtime errors,
+because the frontend is no longer able to decode the data sent from the backend,
+because that data no longer conforms to the standard the frontend expects.
+
+Ideally, we would separate the common types to some files, and use these files in the backend and the frontend.
+Then, the frontend inevitably follows the changes in types from the backend,
+and refuses to compile if the change is too big and a developer has to look at it.
+But, again, Haskell code is almost like Purescript code, but not completely compatible.
+So this is not possible.
+
+### The next best thing
+Well, if we can't use the same files, we have to look for something not that far from that ideal world.
+If we can somehow automatically generate the Purescript code from the Haskell code,
+we could prevent the problem of type difference. We would effectively extend the typesystem from the backend to the frontend.
+
+### `purescript-bridge` to the rescue
+Turns out this idea is not new, and Robert Klotzner has already done it for us, which is quite nice.
+From the docs, purescript-bridge tells us it will write Purescript code from Haskell types,
```

Comment:
@javcasas same here maybe we need to clarify that we are only going to generate types:


```
PureScript types from Haskell types
```

instead of:

```
Purescript code from Haskell types
```